main.tf

```diff
@@ -3,7 +3,7 @@ data "aws_region" "current" {}
 module "ecr" {
   source  = "cloudposse/ecr/aws"
   version = "0.34.0"
-  enabled = var.codepipeline_enabled
+  enabled = var.ecr_enabled || var.codepipeline_enabled
```

Comment:
Why is using `var.codepipeline_enabled` not enough ? Why do we need a new input `var.ecr_enabled` ?

Also, `module.this.enabled` should be part of this too

Reply:
Because there are situations (like ours) where we use ECR but publish to it from GitHub Actions rather than CodePipeline and we don't want the registry destroyed if we set `codepipeline_enabled` to `false`.

Reply:
var.enabled is used in module.this which is set to module.this.enabled which is implicitly used on every cloudposse module via `context` input.

If you override the `enabled` argument on any cloudposse module, then you also need to pass in the `module.this.enabled` in the logic.

i.e.

```hcl
  enabled = module.this.enabled && (var.ecr_enabled || var.codepipeline_enabled)
```

Reply:
cc: @joe-niland ^

Reply:
Updated.